src/main/java/uk/gov/hmcts/reform/sscs/utils/SscsCaseDataUtils.java

```diff
@@ -0,0 +1,149 @@
+            return sscsCaseData.getOtherParties().stream()
+                    .map(CcdValue::getValue)
+                    .map(party -> PartyDetails.builder()
+                            .partyID(party.getId())
+                            .partyType(PartyDetailsUtils.getPartyType(party))
+                            .partyChannel(HearingUtils.getPartyChannel(party.getHearingSubtype()))
+                            .partyName(party.getName() == null ? null : party.getName().getFullName())
+                            .partyRole(party.getRole() == null ? null : party.getRole().getName())
+                            .individualDetails(PartyDetailsUtils.getIndividualDetails(party, sscsCaseData))
+                            .organisationDetails(PartyDetailsUtils.getOrganisationDetails(party))
+                            .unavailabilityDow(null)
+                            .unavailabilityRanges(HearingUtils.getPartyUnavailabilityRange(party.getHearingOptions()))
+                            .build())
+                    .collect(Collectors.toList());
+        }
+        return new ArrayList<>();
+    }
+
+    /**
+     * Get ShvHearingWindow object given SscsCaseData.
+     *
+     * @param caseData SscsCaseData
+     * @return an ShvHearingWindow.
+     */
+    public static HearingWindow getHearingWindow(SscsCaseData caseData) {
+        String hearingWindowStart = null;
+        if (Objects.nonNull(caseData.getEvents())) {
+            Event dwpResponded = caseData.getEvents().stream()
+                    .filter(c -> EventType.DWP_RESPOND.equals(c.getValue().getEventType()))
+                    .findFirst().orElse(null);
+
+            ZonedDateTime dwpResponseDateTime = Optional.ofNullable(dwpResponded)
+                    .map(Event::getValue)
+                    .map(EventDetails::getDateTime)
+                    .orElse(null);
+
+            if (Objects.nonNull(dwpResponseDateTime)) {
+                if (YesNo.isYes(caseData.getUrgentCase())) {
+                    hearingWindowStart = dwpResponseDateTime.plusDays(14).toLocalDate().toString();
+
+                } else {
+                    hearingWindowStart = dwpResponseDateTime.plusDays(28).toLocalDate().toString();
+                }
+            }
+        }
+
+        return HearingWindow.builder()
+                .hearingWindowFirstDate(null)
+                .hearingWindowDateRange(HearingWindowDateRange.builder()
+                        .hearingWindowStartDateRange(hearingWindowStart)
+                        .hearingWindowEndDateRange(null)
+                        .build())
+                .build();
+    }
+
+
+    public static CaseFlags getCaseFlags(SscsCaseData sscsCaseData) {
+        return CaseFlags.builder()
+                .flags(PartyFlagsMapping.getPartyFlags(sscsCaseData).stream()
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList()))
+                .flagAmendUrl(null)
```

Comment:
Should this be null?

Reply:
I've made it empty String. There's nothing to set it to apparent in sscsCaseData, so added a TODO.

Reply:
shouldn't really be doing empty strings.. if there's nothing to add yet don't do anything imo.